src/plots/smith/set_convert.js

```diff
@@ -0,0 +1,204 @@
+ *      axis rotation and direction
+ *
+ * Then, 'g'eometric data is ready to be converted to (x,y).
+ */
+module.exports = function setConvert(ax, smithLayout, fullLayout) {
+    setConvertCartesian(ax, fullLayout);
+
+    switch(ax._id) {
+        case 'x':
+        case 'realaxis':
+            setConvertRadial(ax, smithLayout);
+            break;
+        case 'imaginaryaxis':
+            setConvertAngular(ax, smithLayout);
+            break;
+        case 'realaxis2':
+            setConvertReal(ax, smithLayout);
+            break;
+    }
+};
+
+function setConvertReal(ax, smithLayout) {
+    var subplot = smithLayout._subplot;
+    var radius = subplot.radius;
+
+    ax.setGeometry = function() {
+        ax.c2g = function(v) {
+            var sq = function(x) { return x * x; };
+            var gammaX = function(re) {
+                var denom = sq(re + 1.0);
+                var result = (sq(re) - 1.0) / denom;
+                return result;
+            };
+        // console.log(v, gammaX(v));
+
+            return gammaX(v);
+        };
+
+        ax.g2c = function(v) {
+            return (v + 1.0) / (1.0 - v);
+        };
+
+        ax.g2p = function(v) {
+            return v * radius;
+        };
+
+        ax.c2p = function(v) { return ax.g2p(ax.c2g(v)); };
+    };
+}
+
+function setConvertRadial(ax, smithLayout) {
+    var subplot = smithLayout._subplot;
+
+    ax.setGeometry = function() {
+        var rl0 = ax._rl[0];
+        var rl1 = ax._rl[1];
+
+        var b = subplot.innerRadius;
+        var m = (subplot.radius - b) / (rl1 - rl0);
+        var b2 = b / m;
+
+        var rFilter = rl0 > rl1 ?
+            function(v) { return v <= 0; } :
+            function(v) { return v >= 0; };
+
+        ax.c2g = function(v) {
+            var r = ax.c2l(v) - rl0;
+            return (rFilter(r) ? r : 0) + b2;
+        };
+
+        ax.g2c = function(v) {
+            return ax.l2c(v + rl0 - b2);
+        };
+
+        ax.g2p = function(v) { return v * m; };
+        ax.c2p = function(v) { return ax.g2p(ax.c2g(v)); };
+    };
+}
+
+function toRadians(v, unit) {
+    return unit === 'degrees' ? deg2rad(v) : v;
+}
+
+function fromRadians(v, unit) {
+    return unit === 'degrees' ? rad2deg(v) : v;
+}
+
+function setConvertAngular(ax, smithLayout) {
+    var axType = ax.type;
+
+    if(axType === 'linear') {
+        var _d2c = ax.d2c;
+        var _c2d = ax.c2d;
+
+        ax.d2c = function(v, unit) { return toRadians(_d2c(v), unit); };
+        ax.c2d = function(v, unit) { return _c2d(fromRadians(v, unit)); };
+    }
+
+    // override makeCalcdata to handle thetaunit and special theta0/dtheta logic
+    ax.makeCalcdata = function(trace, coord) {
+        var arrayIn = trace[coord];
+        var len = trace._length;
+        var arrayOut, i;
+
+        var _d2c = function(v) { return ax.d2c(v, trace.thetaunit); };
+
+        if(arrayIn) {
+            if(Lib.isTypedArray(arrayIn) && axType === 'linear') {
```

Comment:
Please remove `linear` condition.

Reply:
Done (but not sure which commit, sorry).